### ISSUE TITLE
Incorrect error report on number of params to macro.

### ISSUE THREAD

**Issue by reltham** [tgjones/HlslTools#4]
Reports "Not enough actual parameters for macro 'RANGED_PARAM_DEFN'."

The macro in question takes 6 params, and 6 params are passed.

Example usage:
RANGED_PARAM_DEFN(float, g_Transparency, Transparency, "Transparency", 0, 1) = 0.5;

Macro definition:
```c
#define RANGED_PARAM_DEFN(type, name, semantic, desc, min, max) type name: semantic <string InstanceCached="true"; string FriendlyName=desc; type UIMin=min; type MinValue=min; type UIMax=max; type MaxValue=max;>
```

I get this for a few different macro definitions that are similar to this one.

**Comment by reltham** [comment]
I believe that when it hits this case it somehow fails to parse correctly afterwards and I get false reports of "Preprocessor directives must appear as the first non-whitespace character on a line." for lines following the macro.

**Comment by tgjones** [comment]
I can't repro this (I've added a test with exactly this macro, and it passes). Perhaps it's related to one of the other issues you logged?

**Comment by reltham** [comment]
I updated the issue text to include the proper full macro definition. It wasn't showing it all because if had <>' in it.

However, I believe you are right that it's caused by other issues, because it doesn't report the error in all cases.  I'll test the latest version and get back to you.

**Comment by reltham** [comment]
I sent you an email with a solution/project and shader files exhibiting this and the other issues. I didn't want to post that here.

**Comment by tgjones** [comment]
Apart from the `##` issue, which we're dealing with separately in #5, this issue was also caused by incorrectly parsing commas in macro arguments, for example:

``` c
PARAM_DEFN(float, MyName, MySemantic, float3(0, 0, 0));
```

I've fixed this in 28e05e3937724279b8f0dadb5ae7d6cb4554a28a and cf78e78bd84267b2cc8ec18ae6323342d835e271.
